controllers/machinedeletionremediation_controller.go

```diff
@@ -134,22 +141,28 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 		return ctrl.Result{RequeueAfter: time.Second}, nil
 	}
 
-	machine, mustExist, err := r.getMachine(remediation)
+	machine, err := r.getMachine(remediation)
 	if err != nil {
-		if apiErrors.IsNotFound(err) {
-			// It is ok to return err == nil here: the Machine does not exist,
-			// either because deleted by MDR or not and the Reconcile has
-			// finished. We return error only if updateConditions failed
-			if mustExist {
-				_, err = r.updateConditions(remediationFailed, remediation)
-			} else {
-				_, err = r.updateConditions(remediationFinished, remediation)
-			}
-			return ctrl.Result{}, err
+		// Handling specific error scenarios. We avoid re-queue by returning nil after updating the
```

Comment:
I don't see returning `nil`?

Reply:
For the custom errors, it returns nil or updateConditions' error

Reply:
ah yes sure, somehow I missed that err is overwritten with the updateCondition err, sorry